interpreter/attributes.go

```diff
@@ -81,6 +81,26 @@ type Attribute interface {
 	Resolve(Activation) (interface{}, error)
 }
 
+// NamespacedAttribute values are a variable within a namespace, and an optional set of qualifiers
+// such as field, key, or index accesses.
+type NamespacedAttribute interface {
+	Attribute
+
+	// CandidateVariableNames returns the possible namespaced variable names for this Attribute in
+	// the CEL namespace resolution order.
```

Comment:
The type suggests this can be a sequence of arbitrary strings, but that's not what your'e expecting.  Consider changing this to something that represents only what you want, e.g. `[]string` where it represents just the components of the FQVN, or use type abstraction.